src/library.js

```diff
@@ -657,7 +657,23 @@ addToLibrary({
 
     {{{ makeSetValue('daylight', '0', 'Number(winterOffset != summerOffset)', 'i32') }}};
 
-    var extractZone = (date) => date.toLocaleTimeString(undefined, {hour12:false, timeZoneName:'short'}).split(' ')[1];
+    var extractZone = (date) => {
+      var timezoneOffset = date.getTimezoneOffset();
+      if (isNaN(timezoneOffset) || timezoneOffset === 0) {
+        return `UTC`;
+      }
+
+      // Why inverse sign?
+      // Read here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getTimezoneOffset
+      var sign = timezoneOffset >= 0 ? "-" : "+";
+
+      var absOffset = Math.abs(timezoneOffset)
+      var hours = String(Math.floor(absOffset / 60)).padStart(2, "0");
+      var minutes = String(absOffset % 60).padStart(2, "0");
+
+      return `UTC${sign}${hours}${minutes}`;
+    }
+
     var winterName = extractZone(winter);
     var summerName = extractZone(summer);
```

Comment:
Can you just pass winterOffset and summerOffset since we already have them?  (i.e. no need to call `getTimezoneOffset` again inside of extractZone).

Reply:
Yes make sense, fixed in https://github.com/emscripten-core/emscripten/pull/22250/commits/f3eaebfdb3381b49aae286461685e804a64f2268